ot3_state_manager/ot3_state_manager/ot3_state_manager.py

```diff
@@ -0,0 +1,262 @@
+    NoPositionProvidedError,
+    PipetteConfigurationError,
+)
+from ot3_state_manager.ot3_state_manager.hardware import (
+    GantryX,
+    GantryY,
+    Gripper,
+    Hardware,
+    Head,
+    LeftPipette,
+    RightPipette,
+    UpdatablePositions,
+)
+from ot3_state_manager.ot3_state_manager.measurable_states import Position
+from ot3_state_manager.ot3_state_manager.pipette_model import PipetteModel
+
+
+class OT3StateManager:
+    """Class to maintain state of OT3.
+
+    Provides properties to get current_position, commanded_position, and
+    encoder_position. Also provides update_position function to update the previously
+    state values.
+    """
+
+    @classmethod
+    def build(
+        cls,
+        left_pipette_model: Optional[PipetteModel],
+        right_pipette_model: Optional[PipetteModel],
+        use_gripper: bool,
+    ) -> OT3StateManager:
+        """Helper function to build an OT3StateManager object.
+
+        Takes optional left pipette and right pipette model names. Passing None for the
+        pipette models names tells the system that there is no pipette attached that
+        slot.
+
+        Pass True for use_gripper parameter to tell the model there is a gripper
+        attached. Pass False if no gripper is attached.
+
+        If you are attaching a high-throughput pipette it must go in the right-side
+        slot.
+        """
+        left_pipette = (
+            LeftPipette(model=left_pipette_model)
+            if left_pipette_model is not None
+            else None
+        )
+        right_pipette = (
+            RightPipette(model=right_pipette_model)
+            if right_pipette_model is not None
+            else None
+        )
+        gripper = Gripper() if use_gripper else None
+
+        return cls(
+            left_pipette=left_pipette, right_pipette=right_pipette, gripper=gripper
+        )
+
+    def __init__(
+        self,
+        left_pipette: Optional[LeftPipette],
+        right_pipette: Optional[RightPipette],
+        gripper: Optional[Gripper],
+    ) -> None:
+
+        if (
+            left_pipette is not None
+            and left_pipette.model == PipetteModel.MULTI_96_1000
+        ):
+            raise PipetteConfigurationError(
+                "High-throughput pipette must be connected to right pipette mount"
+            )
+
+        if (
+            left_pipette is not None
+            and right_pipette is not None
+            and right_pipette.model == PipetteModel.MULTI_96_1000
+        ):
+            raise PipetteConfigurationError(
+                "When using high-throughput pipette you cannot have a pipette in the "
+                "left mount"
+            )
+
+        self.head = Head()
+        self.gantry_x = GantryX()
+        self.gantry_y = GantryY()
+        self.left_pipette = left_pipette
+        self.right_pipette = right_pipette
+        self.gripper = gripper
+
+    def _pos_dict(self) -> Dict[str, Optional[Position]]:
```

Comment:
on the topic of strings - here for instance these really should not be strings. we should live in axis land as much as possible and convert only on boundaries, if there.

Reply:
100b27c5fe94daf9508ee98fcb3e0ecc46012cec